index.ts

```diff
@@ -1,2 +1,46 @@
-import CashBalances from './data';
+import CashBalances, {CashBalanceReport, ReportSection, Account} from './data';
+
+interface ReportAccountTotals {
+  highest_account_name: string
+  lowest_account: string
+  average: any
+}
+
+/**
+ * Takes a report section and returns a flattened list of accounts
+ */
+function getFlattenedListOfAccounts(section: ReportSection): Account[] {
+  return [...section.accounts,
+    ...section.nestedSection[0].accounts, ...section.nestedSection[0].nestedSection[0].accounts,
+    ...section.nestedSection[0].nestedSection[0].nestedSection[0].accounts]
+}
+
+function getAverageAccountBalance(accounts: Account[]):number {
+  let balance = 0;
+
+  accounts.forEach(acc => {
+    balance += acc.value;
+  })
+
+  return balance / accounts.length;
+}
+
+function CalculateAccountTotals(accounts: Account[]): ReportAccountTotals {
+  return {
+    highest_account_name: accounts.sort((a,b) => b.value - a.value).map(acc => a.name)[0] || "",
+    lowest_account: accounts.sort((a,b) => a.value - b.value).map(acc => a.name)[0] || "",
+    average: getAverageAccountBalance(accounts)
+  }
+};
+
+
+/**
+ * Main Program entry. Flattens accounts and then outputs
+ */
+function main(): ReportAccountTotals {
+  let accounts = [...getFlattenedListOfAccounts(CashBalances.assets),
+    ...getFlattenedListOfAccounts(CashBalances.liabilitiesAndEquity)];
+
+  return CalculateAccountTotals(accounts);
```

Comment:
this is pretty dense, how is this for more readable?
```suggestion
  const { assets, liabilitiesAndEquity } = CashBalances;

  const accountList = [ assets, liabilitiesAndEquity ].flatMap( getFlattenedListOfAccounts );

  return CalculateAccountTotals( accountList );
```